glade_forms/test/glade_input_test.dart

```diff
@@ -143,4 +143,48 @@ void main() {
       expect(input.value, equals(100));
     });
   });
+
+  group('Pure test', () {
+    test('ResetToPure', () {
+      final input = GladeInput.intInput(
+        value: 100,
+        useTextEditingController: true,
+      );
+      expect(input.value, equals(100));
+
+      input.updateValue(0, shouldTriggerOnChange: true);
+      expect(input.value, equals(0));
+      input.resetToPure();
+
+      expect(input.isPure, isTrue);
+      // ignore: avoid-duplicate-test-assertions, its not the same as before, its after resetToPure, to test if the initial value is back in.
+      expect(input.value, equals(100));
+    });
+
+    test('SetAsNewPure', () {
+      final input = GladeInput.intInput(
+        value: 100,
+        useTextEditingController: true,
+      );
+
+      // Initial state
+      expect(input.value, equals(100));
+      expect(input.initialValue, equals(100));
+      expect(input.isPure, isTrue);
+
+      // Change the value to make it impure
+      input.updateValue(50);
+      expect(input.value, equals(50));
+      expect(input.isPure, isFalse);
```

Comment:
These lines to change the value to 50 are probably out of scope of this test?

Reply:
That's probably correct, but I just wanted to start from impure value. But yeah, its not needed